pkg/collector/corechecks/system/disk/disk/disk_nix.go

```diff
@@ -8,145 +8,172 @@
+			continue
+		}
+		device := strings.TrimSpace(parts[0])  // e.g. "/dev/sda1"
+		details := strings.TrimSpace(parts[1]) // e.g. `UUID="..." TYPE="ext4" LABEL="root"`
+		match := labelRegex.FindStringSubmatch(details)
+		if len(match) == 2 {
+			// match[1] is everything captured by ([^"]+)
+			c.deviceLabels[device] = match[1]
+		} else {
+			// No label found for this device
+			c.deviceLabels[device] = ""
+		}
 
+	}
 	return nil
 }
 
-func (c *Check) sendPartitionMetrics(sender sender.Sender, usage *disk.UsageStat, tags []string) {
-	// Disk metrics
-	// For legacy reasons,  the standard unit it kB
-	sender.Gauge(fmt.Sprintf(diskMetric, "total"), float64(usage.Total)/1024, "", tags)
-	sender.Gauge(fmt.Sprintf(diskMetric, "used"), float64(usage.Used)/1024, "", tags)
-	sender.Gauge(fmt.Sprintf(diskMetric, "free"), float64(usage.Free)/1024, "", tags)
-	// FIXME(8.x): use percent, a lot more logical than in_use
-	sender.Gauge(fmt.Sprintf(diskMetric, "in_use"), usage.UsedPercent/100, "", tags)
-
-	// Inodes metrics
-	sender.Gauge(fmt.Sprintf(inodeMetric, "total"), float64(usage.InodesTotal), "", tags)
-	sender.Gauge(fmt.Sprintf(inodeMetric, "used"), float64(usage.InodesUsed), "", tags)
-	sender.Gauge(fmt.Sprintf(inodeMetric, "free"), float64(usage.InodesFree), "", tags)
-	// FIXME(8.x): use percent, a lot more logical than in_use
-	sender.Gauge(fmt.Sprintf(inodeMetric, "in_use"), usage.InodesUsedPercent/100, "", tags)
-}
-
-func (c *Check) sendDiskMetrics(sender sender.Sender, ioCounter disk.IOCountersStat, tags []string) {
-	sender.MonotonicCount(fmt.Sprintf(diskMetric, "read_time"), float64(ioCounter.ReadTime), "", tags)
-	sender.MonotonicCount(fmt.Sprintf(diskMetric, "write_time"), float64(ioCounter.WriteTime), "", tags)
-	// FIXME(8.x): These older metrics are kept here for backwards compatibility, but they are wrong: the value is not a percentage
-	sender.Rate(fmt.Sprintf(diskMetric, "read_time_pct"), float64(ioCounter.ReadTime)*100/1000, "", tags)
-	sender.Rate(fmt.Sprintf(diskMetric, "write_time_pct"), float64(ioCounter.WriteTime)*100/1000, "", tags)
-}
-
-// Configure the disk check
-func (c *Check) Configure(senderManager sender.SenderManager, _ uint64, data integration.Data, initConfig integration.Data, source string) error {
-	err := c.CommonConfigure(senderManager, initConfig, data, source)
-	if err != nil {
-		return err
+// NetAddConnection specifies the command used to add a new network connection.
+var NetAddConnection = func(mountType, localName, remoteName, _password, _username string) error {
+	args := []string{}
+	args = append(args, "-t")
+	if mountType == "SMB" {
+		args = append(args, "smbfs")
+	} else {
+		args = append(args, mountType)
 	}
-	return c.instanceConfigure(data)
+	args = append(args, remoteName)
+	args = append(args, localName)
+	cmd := exec.Command("mount", args...)
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	cmd.Stderr = &out
+	err := cmd.Run()
+	return err
```

Comment:
Where does this come from ? I couldn't find this in the Python version

Reply:
This is what I commented above. Non-windows version of the `create_mounts` config. Removed and maybe it'll be implemented in a future PR